File: movie-infrastructures/storage/src/main/java/com/movie/storage/movie/mapper/TheaterMapper.java
package com.movie.storage.movie.mapper;

import com.movie.domain.movie.domain.Theater;
import com.movie.storage.movie.entity.TheaterEntity;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class TheaterMapper {

    public static Theater from(TheaterEntity entity) {
        return Theater.of(entity.getId(), entity.getName());
    }

}

File: data/src/main/kotlin/kr/spartacodingclub/cinema/data/mapper/TheaterMapper.kt
package kr.spartacodingclub.cinema.data.mapper

import kr.spartacodingclub.cinema.core.domain.Theater
import kr.spartacodingclub.cinema.data.entity.TheaterEntity
import org.mapstruct.Mapper
import org.mapstruct.factory.Mappers

@Mapper
interface TheaterMapper {
    companion object {
        val INSTANCE: TheaterMapper = Mappers.getMapper(TheaterMapper::class.java)
    }

    fun toTheater(entity: TheaterEntity): Theater
    fun toTheaterEntity(domain: Theater): TheaterEntity
}
